app/README.md

```diff
@@ -0,0 +1,57 @@
+# App modules
+
+App modules are subsections of the main site, dealing with a specific area , e.g locations or moves. They are constructed using [Express](https://expressjs.com) [routers](https://expressjs.com/en/api.html#router), [param middleware](https://expressjs.com/en/api.html#router.param) and [shared middleware](https://expressjs.com/en/api.html#router.use).
+
+They can have a variety of different parts, in either a root file, or a folder with multiple files, and exposed via an `index.js` in the folder.
+
+- `controllers` - Express controllers
+- `constants` - Predefined default settings for a module, including routing paths, actions, and other defaults
+- `fields` - Form-wizard field definitions
+- `middleware` - Express middleware for the module, rendering against the views
+- `steps` - Form-wizard step definitions
+- `validators` - Form-wizard field validator functions
+- `views` - Nunjucks templates`
+- `index / routing` - Main entrypoint for the module, tying together all the parts of the module in the form of an Express router
+
+## Form Wizard
+
+Note: Multi-page forms are provided using the [HMPO Form Wizard](https://github.com/HMPO/hmpo-form-wizard) and have a different setup and lifecycle. Please refer to the linked documentation for more information and not the app module specific details below.
+
+## Routing
+
+The index file of each app module is used to setup any routes or subroutes that this module will serve through Express. The [standard Express middleware pattern](https://expressjs.com/en/guide/using-middleware.html) is used here with a variety of middleware using and transforming the request object, with a final method that will usually render content.
+
+## Middleware
+
+There are a variety of generic middleware that are provided from within the [common](../common/middleware) folder
+
+## Common Middleware
+
+- `permissions` - secures route access based upon user permissions
+- `setLocations` - sets locations on the current request
+- `setPrimaryNaviation` - sets the primaryNavigation based upon the current users permissions
+- `setUser` - sets the user on the current request from the session
+
+### Common Collections Middleware
+
+- `redirectDefaultQuery` - redirect with a default query if provided
+- `redirectView` - redirect to a path of the format `timePeriod/date/location/view?query`
+- `setActions` - sets potential actions during the routing setup phase
+- `setContext` - sets the `context` used by [i18n]() to specific content within a generalised template
+- `setDateRange` - sets the `dateRange` used by the weekly views
+- `setPagination` - sets the pagination object used for day/week pagination. Note: This is not for numeric pagination of table data.
+
+## Orchestration
```

Comment:
I'm not really happy with this name, but I couldn't think of another grouping for these three kinds of middleware